### PR TITLE
Fix interaction of Dragon Darts and Berserk

### PR DESCRIPTION
**Pull request by singiamtel**
As suggested by: [Dragon Darts and Berserk](https://github.com/smogon/pokemon-showdown/projects/3#card-61614678)

Wow, this one was quite tricky to find.

### CODE REVIEW

File: sim/battle-actions.ts [hunk 1]
```diff
@@ -942,7 +943,13 @@ export class BattleActions {
 		}
 
 		// smartTarget messes up targetsCopy, but smartTarget should in theory ensure that targets will never fail, anyway
-		if (move.smartTarget) targetsCopy = targets.slice(0);
+		if (move.smartTarget) {
```
Comment (singiamtel) [inline sim/battle-actions.ts:946]: The issue was in this copy, reverting the safety check done for smartTarget earlier

File: sim/battle-actions.ts [hunk 2]
```diff
@@ -842,6 +842,7 @@ export class BattleActions {
 			move.hit = hit;
 			if (move.smartTarget && targets.length > 1) {
 				targetsCopy = [targets[hit - 1]];
+				damage = [damage[hit - 1]];
```
Comment (singiamtel) [inline sim/battle-actions.ts:845]: This was in practice not breaking anything, but still weird to have the trailing zero in the damage array

**Review by DaWoblefet — APPROVED** [review]

**Comment by singiamtel** [comment]
How should we proceed? I didn't want to push my PRs as I understand you'd be busy with SV release.

#8990, #8984 and this one are ready for review/merge

**Comment by singiamtel** [comment]
Accidentaly closed it trying to sync my repo with upstream

**Comment by DaWoblefet** [comment]
Thanks singiamtel!

**Pull request merged**